### ISSUE TITLE
pbecho.py and pbechoclient.py fail to give a proper example of exception propagation

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3673]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#3673|
|Type|defect|
|Created|2009-02-24 20:07:05Z|

By following the poor practice of defining a bunch of classes in `__main__`, `pbecho.py` falls into the most common trap this practice lays - that the exception class it tries to send to `pbechoclient.py` cannot be handled reasonably by the client.

Classes and functions should never be defined in `__main__`.  The examples ought not to be an exception to this rule, and in this case, doing things wrong actually leads to a broken example.  As if showing end users, _in an example of how to use Twisted_, the **wrong** way to use Twisted weren't bad enough by itself.

Attachments:

* [fix.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/031/0314d5a7190ae6d6c642d1bbb6f94f910ab01b9d/653b8a47b62a3d2b9e03df768e3b54c79b513045.patch) (757 bytes) - added by exarkun on 2009-02-24 20:09:10Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__3673 3673
type__defect defect
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__documentation documentation
time__1235506025000000 1235506025000000
changetime__1235922352000000 1235922352000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun removed owner|
|-|-|

I wrote this simple patch while investigating the user report of this problem.  It's kind of a fix, but perhaps these examples should be unit tested and suchlike as well.  Actually no, that's scope creep.  This example should be fixed to _work_ now, and a more significant overhaul made later on so as not to block this simple fix.

**Comment by twisted-trac** [comment]
|[<img alt="jerub's avatar" src="https://avatars.githubusercontent.com/u/315461?s=50" width="50" height="50">](https://github.com/jerub)<a name="note_2"></a>|@jerub set owner to @exarkun|
|-|-|

I agree, please merge this fix.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun set status to `closed`|
|-|-|

(In [26325]) Apply patch fixing pbecho server example

Author: exarkun
Reviewer: jerub
Fixes: [#3673](https://github.com/twisted/twisted/issues/3673)

Change the PB echo server example so that it doesn't use names defined in
`__main__` which cannot easily be sent over a PB connection.  This fixes the
behavior of the example which demonstrates failures.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
